### ISSUE TITLE
block_log got corrupted (?) Assert Exception size.value < MAX_ARRAY_ALLOC_SIZE

### ISSUE THREAD

**Issue by Gandalf-the-Grey** [steemit/steem#1255]
I used to get `block_log` from running instance to re-use it for `--replay` on new nodes, it worked flawlessly for months, but today I've noticed troubles, `steemd` assert with:
```
steem@seed:~$ ./bin/steemd-v0.19.0-full --flush=1000000 -d fullnode --replay 2>&1 | tee -a fullnode.log
------------------------------------------------------

            STARTING STEEM NETWORK

------------------------------------------------------
initminer public key: STM8GC13uCZbP44HzMLV6zPZGwVQ8Nt4Kji8PapsPiNq1BK153XTX
chain id: 0000000000000000000000000000000000000000000000000000000000000000
blockchain version: 0.19.0
------------------------------------------------------
2693628ms th_a       main.cpp:172                  main                 ] parsing options
2693628ms th_a       main.cpp:174                  main                 ] initializing node
2693628ms th_a       main.cpp:176                  main                 ] initializing plugins
2693628ms th_a       application.cpp:1140          initialize_plugins   ] Initializing plugin account_by_key
2693628ms th_a       account_by_key_plugin.cpp:245 plugin_initialize    ] Initializing account_by_key plugin
2693628ms th_a       application.cpp:1140          initialize_plugins   ] Initializing plugin account_history
2693628ms th_a       application.cpp:1140          initialize_plugins   ] Initializing plugin follow
2693628ms th_a       follow_plugin.cpp:360         plugin_initialize    ] Intializing follow plugin
2693628ms th_a       application.cpp:1140          initialize_plugins   ] Initializing plugin market_history
2693628ms th_a       market_history_plugin.cpp:177 plugin_initialize    ] market_history: plugin_initialize() begin
2693628ms th_a       market_history_plugin.cpp:192 plugin_initialize    ] bucket-size [15,60,300,3600,86400]
2693629ms th_a       market_history_plugin.cpp:193 plugin_initialize    ] history-per-size 1000
2693629ms th_a       market_history_plugin.cpp:195 plugin_initialize    ] market_history: plugin_initialize() end
2693629ms th_a       application.cpp:1140          initialize_plugins   ] Initializing plugin raw_block
2693629ms th_a       application.cpp:1140          initialize_plugins   ] Initializing plugin tags
2693629ms th_a       tags_plugin.cpp:525           plugin_initialize    ] Intializing tags plugin
2693629ms th_a       application.hpp:105           register_api_factory ] name: tag_api
2693629ms th_a       application.cpp:1140          initialize_plugins   ] Initializing plugin witness
2693629ms th_a       main.cpp:179                  main                 ] starting node
2693629ms th_a       application.cpp:256           startup              ] shared_file_size is 107374182400 bytes
2693629ms th_a       application.hpp:105           register_api_factory ] name: login_api
2693629ms th_a       application.hpp:105           register_api_factory ] name: database_api
2693629ms th_a       application.hpp:105           register_api_factory ] name: network_node_api
2693629ms th_a       application.hpp:105           register_api_factory ] name: network_broadcast_api
2693629ms th_a       application.cpp:274           startup              ] Starting Steem node in write mode.
2693629ms th_a       application.cpp:297           startup              ] Replaying blockchain on user request.
2693629ms th_a       database.cpp:151              reindex              ] Reindexing Blockchain
2693687ms th_a       block_log.cpp:130             open                 ] Log is nonempty
2693687ms th_a       block_log.cpp:161             open                 ] Index is empty
2693687ms th_a       block_log.cpp:282             construct_index      ] Reconstructing Block Log Index...
2746681ms th_a       block_log.cpp:305             construct_index      ] 10 assert_exception: Assert Exception
size.value < MAX_ARRAY_ALLOC_SIZE:
    {}
    th_a  raw.hpp:239 unpack
Error unpacking field permlink
    {"field":"permlink"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::vote_operation
    {"type":"steemit::protocol::vote_operation"}
    th_a  raw.hpp:534 unpack
Error unpacking field operations
    {"field":"operations"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_transaction
    {"type":"steemit::protocol::signed_transaction"}
    th_a  raw.hpp:534 unpack
Error unpacking field transactions
    {"field":"transactions"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_block
    {"type":"steemit::protocol::signed_block"}
    th_a  raw.hpp:534 unpack
2746681ms th_a       database.cpp:144              open                 ] 10 assert_exception: Assert Exception
size.value < MAX_ARRAY_ALLOC_SIZE:
    {}
    th_a  raw.hpp:239 unpack
Error unpacking field permlink
    {"field":"permlink"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::vote_operation
    {"type":"steemit::protocol::vote_operation"}
    th_a  raw.hpp:534 unpack
Error unpacking field operations
    {"field":"operations"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_transaction
    {"type":"steemit::protocol::signed_transaction"}
    th_a  raw.hpp:534 unpack
Error unpacking field transactions
    {"field":"transactions"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_block
    {"type":"steemit::protocol::signed_block"}
    th_a  raw.hpp:534 unpack
rethrow
    {}
    th_a  block_log.cpp:305 construct_index
2746681ms th_a       database.cpp:144              open                 ] data_dir: /home/steem/fullnode/blockchain shared_mem_dir: /run/fullnode shared_file_size: 107374182400
2746681ms th_a       application.cpp:390           startup              ] 10 assert_exception: Assert Exception
size.value < MAX_ARRAY_ALLOC_SIZE:
    {}
    th_a  raw.hpp:239 unpack
Error unpacking field permlink
    {"field":"permlink"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::vote_operation
    {"type":"steemit::protocol::vote_operation"}
    th_a  raw.hpp:534 unpack
Error unpacking field operations
    {"field":"operations"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_transaction
    {"type":"steemit::protocol::signed_transaction"}
    th_a  raw.hpp:534 unpack
Error unpacking field transactions
    {"field":"transactions"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_block
    {"type":"steemit::protocol::signed_block"}
    th_a  raw.hpp:534 unpack
rethrow
    {}
    th_a  block_log.cpp:305 construct_index
rethrow
    {"data_dir":"/home/steem/fullnode/blockchain","shared_mem_dir":"/run/fullnode","shared_file_size":"107374182400"}
    th_a  database.cpp:144 open

    {"data_dir":"/home/steem/fullnode/blockchain","shared_mem_dir":"/run/fullnode"}
    th_a  database.cpp:199 reindex
2746681ms th_a       application.cpp:1012          startup              ] 10 assert_exception: Assert Exception
size.value < MAX_ARRAY_ALLOC_SIZE:
    {}
    th_a  raw.hpp:239 unpack
Error unpacking field permlink
    {"field":"permlink"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::vote_operation
    {"type":"steemit::protocol::vote_operation"}
    th_a  raw.hpp:534 unpack
Error unpacking field operations
    {"field":"operations"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_transaction
    {"type":"steemit::protocol::signed_transaction"}
    th_a  raw.hpp:534 unpack
Error unpacking field transactions
    {"field":"transactions"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_block
    {"type":"steemit::protocol::signed_block"}
    th_a  raw.hpp:534 unpack
rethrow
    {}
    th_a  block_log.cpp:305 construct_index
rethrow
    {"data_dir":"/home/steem/fullnode/blockchain","shared_mem_dir":"/run/fullnode","shared_file_size":"107374182400"}
    th_a  database.cpp:144 open

    {"data_dir":"/home/steem/fullnode/blockchain","shared_mem_dir":"/run/fullnode"}
    th_a  database.cpp:199 reindex
rethrow
    {}
    th_a  application.cpp:390 startup
2746681ms th_a       main.cpp:213                  main                 ] Exiting with error:
10 assert_exception: Assert Exception
size.value < MAX_ARRAY_ALLOC_SIZE:
    {}
    th_a  raw.hpp:239 unpack
Error unpacking field permlink
    {"field":"permlink"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::vote_operation
    {"type":"steemit::protocol::vote_operation"}
    th_a  raw.hpp:534 unpack
Error unpacking field operations
    {"field":"operations"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_transaction
    {"type":"steemit::protocol::signed_transaction"}
    th_a  raw.hpp:534 unpack
Error unpacking field transactions
    {"field":"transactions"}
    th_a  raw.hpp:293 operator()
error unpacking steemit::protocol::signed_block
    {"type":"steemit::protocol::signed_block"}
    th_a  raw.hpp:534 unpack
rethrow
    {}
    th_a  block_log.cpp:305 construct_index
rethrow
    {"data_dir":"/home/steem/fullnode/blockchain","shared_mem_dir":"/run/fullnode","shared_file_size":"107374182400"}
    th_a  database.cpp:144 open

    {"data_dir":"/home/steem/fullnode/blockchain","shared_mem_dir":"/run/fullnode"}
    th_a  database.cpp:199 reindex
rethrow
    {}
    th_a  application.cpp:390 startup
```
reproduced multiple times (by downloading `block_log` from that instance each time)
for a `block_log` backed up earlier (from same instance in same way) it works ok
`steemd` instance that is source of that broken `block_log` is still up and running

**Comment by VIM-Arcange** [comment]
I confirm! I had exactly the behavior today when replaying my full node.

**Comment by mvandeberg** [comment]
Were you copying the block log from running steemd node or did you shut them down and then copy it?

This looks to be a corrupted block log. I have not seen this behavior other places where we deserialize blocks, I do not believe there is a problem with the serialization logic and that it is likely a corrupted block log.

**Comment by Gandalf-the-Grey** [comment]
Yes, it was copied from running `steemd`. While it is a bit of a hack, since HF16 I was able to use it to serve to other nodes "always up to date" `block_log` file.
Node that's producing this `block_log` was still running, getting blocks, etc. I was able to shut that node down cleanly and reproduce it on a copied `block_log`. Bringing node back up went without issues as long as `block_log.index` was there. After removing it replaying original instance of course failed at reconstructing index. 
That's the only case like this among all my nodes. Tried on others multiple times (by taking `block_log` from live instances) - worked every time.
I suppose some random `block_log` corruption then, hopefully by external causes, but I'm afraid about same issue at @VIM-Arcange's node mentioned above.
